pkg/controllermanager/controller/shoot/maintenance/reconciler.go

```diff
@@ -299,6 +299,20 @@ func (r *Reconciler) reconcile(ctx context.Context, log logr.Logger, shoot *gard
 		return err
 	}
 
+	// if the maintenance patch is not required and the last maintenance operation state is failed,
+	// this means the maintenance was retried and succeeded. Alternatively, changes could have been made
+	// outside of the maintenance window to fix the maintenance error. In either case, remove the failed state.
+	if !requirePatch && shoot.Status.LastMaintenance != nil && shoot.Status.LastMaintenance.State == gardencorev1beta1.LastOperationStateFailed {
```

Comment:
The changes look good but I wonder if it wasn't easier to reset the status every time we begin with the maintenance loop. 

This might require a few more patches, but is easy to grasp and follow the example of the shoot reconciliation, that resets the `lastOperation` (see [here](https://github.com/gardener/gardener/blob/79c8111320e5237f15f54030b3d53aaa40ab57b6/pkg/gardenlet/controller/shoot/shoot/reconciler.go#L298)).

At the moment, it's yet another corner case that requires explanation in an already nested and intricate controller logic. 

WDYT?

Reply:
>The changes look good but I wonder if it wasn't easier to reset the status every time we begin with the maintenance loop.

This is not what we want, when the feature was added, the idea was to update the status only if something is changed, so we could look at what was changed during the maintenance, if it is reset in every reconciliation, what was changed during the maintenance will stay in status only for one day.

Reply:
Thanks for the explanation.